File: Examples/PackageDiagramDiff/Kotlin/Before/Banking/InMemoryAccountRepository.kt
/** An in-memory `AccountRepository`. Part of the `Banking` module; depends on `Core`. */
class InMemoryAccountRepository : AccountRepository {
    private val storage: MutableMap<String, Account> = mutableMapOf()

    override fun find(id: String): Account? = storage[id]

    override fun save(account: Account) {
        storage[account.id] = account
    }
}

File: Examples/PackageDiagramDiff/Dart/Before/Banking/in_memory_account_repository.dart
/// An in-memory `AccountRepository`. Part of the `Banking` module; depends on `Core`.
class InMemoryAccountRepository implements AccountRepository {
  final Map<String, Account> _storage = {};

  @override
  Account? find(String id) => _storage[id];

  @override
  void save(Account account) {
    _storage[account.id] = account;
  }
}
